### ISSUE TITLE
Can't polyfill intl-pluralrules with 'fr' locale-data 

### ISSUE THREAD

**Issue by lathou** [formatjs/formatjs#2482]
## Which package?
@formatjs/intl-pluralrules

## Describe the bug
Using "fr" locale-data from @formatjs/intl-pluralrules/locale-data/fr create an error on old browser like Chrome 70 : 
```
Error: [@formatjs/intl Error FORMAT_ERROR] Error formatting relative time. 
Locale: ReferenceError: e is not defined
MessageID: undefined
Default Message: undefined
Description: undefined 
```

## To Reproduce
Use FormattedRelativeTime component, with these imports, on old browser :
```
import '@formatjs/intl-getcanonicallocales/polyfill';        //v1.5.3
import '@formatjs/intl-locale/polyfill';        //v2.4.12
import '@formatjs/intl-pluralrules/polyfill';       //v4.0.2
import '@formatjs/intl-pluralrules/locale-data/fr';
import '@formatjs/intl-numberformat/polyfill';        //v6.1.2
import '@formatjs/intl-numberformat/locale-data/fr'; 
import '@formatjs/intl-relativetimeformat/polyfill';        //v8.0.2
import '@formatjs/intl-relativetimeformat/locale-data/fr';
import { IntlProvider } from 'react-intl';        //v5.10.12
```

Using `import '@formatjs/intl-pluralrules/locale-data/en'` instead of `import '@formatjs/intl-pluralrules/locale-data/fr'` fix the error.

## Expected behavior
The component <FormattedRelativeTime/> renders the correct value and do not throw error on Chrome 70.

## Desktop:

- OS: UBUNTU
- Browser: Chrome
- Version 70

**Comment by longlho** [comment]
Do u have a repro or stack trace?

**Comment by lathou** [comment]
Here is the error with stack trace :

```
utils.js?d555:19 Error: [@formatjs/intl Error FORMAT_ERROR] Error formatting relative time. 
Locale: ReferenceError: e is not defined
MessageID: undefined
Default Message: undefined
Description: undefined 

    at new MessageFormatError (error.js?12e1:50)
    at formatRelativeTime (relativeTime.js?0423:27)
    at eval (relative.js?caad:142)
    at updateContextConsumer (react-dom.development.js?61bb:18304)
    at beginWork (react-dom.development.js?61bb:18661)
    at beginWork$1 (react-dom.development.js?61bb:23179)
    at performUnitOfWork (react-dom.development.js?61bb:22157)
    at workLoopSync (react-dom.development.js?61bb:22130)
    at performSyncWorkOnRoot (react-dom.development.js?61bb:21756)
    at eval (react-dom.development.js?61bb:11089)
```

And here is when I console.log the error, it helped me to find where the problem was coming from :

```
ReferenceError: e is not defined
    at Object.fn (fr.js?7ea8:8)
    at PluralRuleSelect (index.js?7107:21)
    at Module.ResolvePlural (ResolvePlural.js?c68a:24)
    at PluralRules.select (index.js?7107:66)
    at Module.PartitionRelativeTimePattern (PartitionRelativeTimePattern.js?ffe8:51)
    at RelativeTimeFormat.format (index.js?8a87:30)
    at formatRelativeTime (relativeTime.js?0423:23)
    at eval (relative.js?caad:142)
    at updateContextConsumer (react-dom.development.js?61bb:18304)
    at beginWork (react-dom.development.js?61bb:18661)
```

**Comment by dbendaou** [comment]
Same issue on EDGE 18, Safari 12.1, chrome 69 and Firefox 52
Here's the stack trace on safari (sorry for the poor quality it was on browserStack)
<img width="552" alt="Screenshot 2021-01-06 at 16 49 24" src="https://user-images.githubusercontent.com/15218861/103788567-42deee00-503f-11eb-9642-a9b086883869.png">
<img width="766" alt="Screenshot 2021-01-06 at 16 52 25" src="https://user-images.githubusercontent.com/15218861/103788851-9bae8680-503f-11eb-9c5c-d17f621cba49.png">


import in my local manager service look like 
```javascript 
import { shouldPolyfill as shouldPolyfillPluralRules } from '@formatjs/intl-pluralrules/should-polyfill';

[...]
  if (shouldPolyfillPluralRules()) {
    await import('@formatjs/intl-locale/polyfill');
    await import('@formatjs/intl-getcanonicallocales/polyfill');
    await import('@formatjs/intl-pluralrules/polyfill');
  }
```

**Comment by longlho** [comment]
Thanks for the stack trace. It looks like `e` was introduced in cldrv38 specifically for French. I'll make a corresponding PR in our plural compiler to address this.

**Comment by dbendaou** [comment]
Thrilled! Thanks 🤗

**Comment by dbendaou** [comment]
Could we let it open until it's fixed?

**Comment by longlho** [comment]
It's fixed in latest version 4.0.5

**Comment by dbendaou** [comment]
Thank you for the quick fix !
